pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -1228,7 +1228,7 @@ def run_backward_match(self):
                         condition = True
 
                         for back_match in match_backward:
-                            if back_match not in new_matches_scenario_match:
+                            if back_match not in new_matches_scenario_match:  # pragma: no cover
```

Comment:
How come this line would not be covered?

Reply:
It is an edge case that is in the proof but I don't have a concrete example to test it. It would require a big circuit and a big template (that I need to find), that would probably take at least 20-30s to run as well. I don't think it is worth.

Reply:
Got it. Maybe worth making a comment on this in the code.